src/codegate/pipeline/secrets/secrets.py

```diff
@@ -312,8 +312,8 @@ class SecretUnredactionStep(OutputPipelineStep):
     """Pipeline step that unredacts protected content in the stream"""
 
     def __init__(self):
-        self.redacted_pattern = re.compile(r"REDACTED<\$([^>]+)>")
-        self.marker_start = "REDACTED<$"
+        self.redacted_pattern = re.compile(r"REDACTED<(\$?[^>]+)>")
```

Comment:
As with all regexes, this will start getting harder to maintain with time. Mind adding a couple of examples expected patterns?